src/Features/Core/Portable/Diagnostics/DiagnosticHelper.cs

```diff
@@ -108,6 +108,59 @@ static string EncodeIndices(IEnumerable<int> indices, int additionalLocationsLen
             }
         }
 
+        public static Diagnostic CreateWithUnnecessaryLocations(
+            DiagnosticDescriptor descriptor,
+            Location location,
+            ReportDiagnostic effectiveSeverity,
+            IEnumerable<Location> additionalLocations,
+            IEnumerable<Location> unnecessaryAdditionalLocations,
+            params object[] messageArgs)
+        {
+            var allAdditionalLocations = ImmutableArray.CreateBuilder<Location>();
```

Comment:
this isn't freed afaict...

Reply:
Freed? This isn't pooled.

Reply:
should it be?  that's our general pattern for tehse guys.  that way we can at least recover the builer for future use.  i.e. ArrayBuilder.GetInstance.